### PR TITLE
Add jsdocs for Server class

### PR DESCRIPTION
**Pull request by paulbellamy**
Fixes https://github.com/stellar/js-soroban-client/issues/20

# TODO

- [x] Add examples for each method
- [x] Update examples in docs/reference
  - [x] Expose a global SorobanClient on laboratory.stellar.org (or remove the reference in the docs)
- [ ] Figure out how to emit docs for the response types -> moved to https://github.com/stellar/js-soroban-client/issues/28

### CODE REVIEW

**Comment by paulbellamy** [comment]
cc @piyalbasu how hard would exposing a global SorobanClient library on laboratory.stellar.org be? Worth it for the docs to be able to say you can run stuff there, or nah? We do that for js-stellar-sdk, but I'd rather link to a replit or somesuch tbh.
Thoughts @piyalbasu  & @leighmcculloch ?

**Comment by leighmcculloch** [comment]
> cc @piyalbasu how hard would exposing a global SorobanClient library on laboratory.stellar.org be? Worth it for the docs to be able to say you can run stuff there, or nah? We do that for js-stellar-sdk, but I'd rather link to a replit or somesuch tbh.
> Thoughts @piyalbasu & @leighmcculloch ?

Is the goal to tell people to open the browser console and run things there? Or to build a UI in the lab? I'd avoid telling people to use the browser console, people trick other people to run commands in the console, stealing cookies, etc, so we should probably not build that into peoples workflows.

**Comment by paulbellamy** [comment]
afaict the goal is basically give them an easy pre-set-up repl to follow along in. I agree, browser console sets a bad security precedent. Is there a better place to link people to?

**Comment by leighmcculloch** [comment]
@tyvdh Do you have some ideas for ☝️? My first thought is runkit.

**Comment by tyvdh** [comment]
I know @ElliotFriend has been doing a lot of work in the docs for having runnable inline code with a micro frontend. Seems like this could be similar. Something like a Replit, Codesandbox, Glitch, etc.

@ElliotFriend would be curious to see what the current state of that project is and if we could do something similar here for the Soroban client.

**Comment by ElliotFriend** [comment]
Hey, everyone!

That project got sidelined a bit when we pivoted to Soroban for SQ5. But it's been coming back into my main field of view since we now have Soroban Quest (mostly) behind us. As I left it, I had done a pretty thorough comparison of a **bunch** of different options (codesandbox, replit, codepen, stackblitz, codemirror, glitch, and gitpod). Of all the options, it seems like **codesandbox** best matches what I was looking for. There were pretty decent options for customizing and configuring what is displayed in the embed (specific files, paths, previews, console, etc.).

I also had a PoC locally (that I just pushed to my fork this morning) with a few different embeds to test things out. It was pretty easy to get working, and you can [see the embedded bits here](https://github.com/ElliotFriend/stellar-docs/commit/51cb3693a3c5ae4d0e1b58fe5e8ca987b1866317#diff-80f49163d3a0f9b31da4997f5dd6da0552bb70fb5cc2715c158c9caeef14350fR13).

I don't really know how to stand up a preview of the branch, but here's a screenshot of what it was looking like

Screenshot from 2022-10-26 11-18-36

If it's more of a focus on the JS code that's necessary, (as opposed to building a *slightly* more comprehensive web app like for `stellar-docs`), we can certainly embed runkit with a pre-set-up environment, too. I love the way our "code mode" stuff on SQ turned out!

**Comment by ElliotFriend** [comment]
For the sake of complete-ness, here's how our embedded runkit looks:

Screenshot from 2022-12-14 11-30-36

**Comment by paulbellamy** [comment]
Ok. Those look fantastic. I'll leave out the bit about laboratory, and we can link to wherever these are accessible.

**Comment by paulbellamy** [comment]
Going to break out the `SorobanRpc.*Response` types emission fix to a separate issue to get this merged as-is (still better than what we have).

File: docs/reference/examples.md [hunk 1]
```diff
@@ -2,150 +2,95 @@
...
 
 const receiverPublicKey = 'GAIRISXKPLOWZBMFRPU5XRGUUX3VMA3ZEWKBM5MSNRU3CHV6P4PYZ74D';
 
-// Configure StellarSdk to talk to the horizon instance hosted by Stellar.org
+const contractId = '0000000000000000000000000000000000000000000000000000000000000001';
+
+// Configure SorobanClient to talk to the horizon instance hosted by Stellar.org
 // To use the live network, set the hostname to 'horizon.stellar.org'
-const server = new StellarSdk.Server('https://horizon-testnet.stellar.org');
+const server = new SorobanClient.Server('https://soroban-rpc-testnet.stellar.org');
```
Comment (Shaptic) [inline docs/reference/examples.md:40]: sanity check: is this url correct?

File: src/server.ts [hunk 2]
```diff
@@ -56,19 +54,62 @@ export class Server {
...
+   * const key = xdr.ScVal.scvSymbol("counter");
+   * server.getContractData(contractId, key).then(data => {
+   *   console.log("value:", data.xdr);
+   *   console.log("lastModified:", data.lastModifiedLedgerSeq);
+   *   console.log("latestLedger:", data.latestLedger);
+   * });
+   *
+   * Allows you to directly inspect the current state of a contract. This is a backup way to access your contract data which may not be available via events or simulateTransaction.
+   *
+   * @deprecated Use {@link Server#getLedgerEntry} instead.
```
Comment (Shaptic) [inline src/server.ts:108]: Can we make sure there's an item somewhere to drop this? It's too early to have long-lived deprecations :joy:

File: src/server.ts [hunk 3]
```diff
@@ -101,6 +237,49 @@ export class Server {
     );
   }
 
+  /**
+   * Submit a real transaction to the Stellar network. This is the only way to make changes "on-chain".
+   * Unlike Horizon, Soroban-RPC does not wait for transaction completion. It simply validates the transaction and enqueues it. Clients should call {@link Server#getTransactionStatus} to learn about transaction success/failure.
```
Comment (Shaptic) [inline src/server.ts:242]: wrap?
```suggestion
   * Unlike Horizon, Soroban-RPC does not wait for transaction completion. 
   * It simply validates the transaction and enqueues it. Clients should call 
   * {@link Server#getTransactionStatus} to learn about transaction success/failure.
```

File: src/server.ts [hunk 4]
```diff
@@ -101,6 +237,49 @@ export class Server {
...
+   * const sourceKeypair = SorobanClient.Keypair.fromSecret(sourceSecretKey);
+   * transaction.sign(sourceKeypair);
+   *
+   * server.sendTransaction(transaction).then(result => {
+   *   console.log("id:", result.id);
+   *   console.log("error:", result.error);
+   * });
+   *
+   * @param {Transaction | FeeBumpTransaction} transaction - The transaction to submit.
+   * @returns {Promise<SorobanRpc.SendTransactionResponse>} Returns a promise to the {@link SorobanRpc.SendTransactionResponse} object with the transaction id, status, and any error if available.
```
Comment (Shaptic) [inline src/server.ts:281]: same here: wrap?
```suggestion
   * @returns {Promise<SorobanRpc.SendTransactionResponse>} Returns a promise to the 
   *     {@link SorobanRpc.SendTransactionResponse} object with the transaction id, status, 
   *     and any error if available.
```

**Review by Shaptic — APPROVED** [review]
noice docs

File: docs/reference/examples.md (hunk 1 above)
Comment (paulbellamy) [reply docs/reference/examples.md:40]: It doesn't exist, no. But there's no public one, so I wasn't sure where to set it to tbh. Maybe we should just use localhost for now... 🤔

File: src/server.ts (hunk 2 above)
Comment (paulbellamy) [reply src/server.ts:108]: https://github.com/stellar/soroban-tools/issues/316

**Pull request merged**